### ISSUE TITLE
WAFileMetadataLibrary does not conserve folder hierarchy on windows

### ISSUE THREAD

**Issue by jecisc** [SeasideSt/Seaside#941]
I imported files with the method `#recursivelyAddAllFilesIn:`. With that, the folder hierarchy is included in the metadata of each files. The problem is that, when I deploy them (`#deploy` method), the hierarchy is not preserved. 

The problem seems to come from `#asFileAndFoldername:`.

In this method we get the pathSeparator for the current platform. For windows this is `\`. Then there is this code:

```smalltalk
filename := filename replaceAll: $/ with: pathSeparator first.
	foldername := (filename includes: $/)
		ifTrue: [ self name , pathSeparator, (filename copyUpToLast: pathSeparator first) ]
		ifFalse: [ self name ].
```

On windows we will never get in the true branch since we replaced all the `/` by a `\`

**Comment by gcotelli** [comment]
`#asFileAndFoldername:` also has a undesired side effect. 
`filename replaceAll: $/ with: pathSeparator first.` will mutate the given string.  Maybe a copyReplaceAll:with: is better in this case.

This can explain some weird test case failures I'm seeing on Windows, testing the deployment of some subclasess of WAFileMetadataLibrary. Given this string is the file path in the method defining the resource, this will mutate something that is in a method literal.

**Comment by jecisc** [comment]
Maybe we could add an appveyor build.

**Comment by marschall** [comment]
The way I understand the code `filepath` should always use `$/` and the conversion to the native separator is supposed to happen in `#asFileAndFoldername:`

`#replaceAll:with:` ↔ `#copyReplaceAll:with:` seems to be a valid concern

**Comment by gcotelli** [comment]
On Windows it doesn't work right because, just after the ifTrue: $/ is replaced in `filename` by the platform path separator, so `(filename includes: $/)` ends up being false, and the folder information is lost in `foldername`.

I think changing `(filename includes: $/)` by `(filename includesSubstring: pathSeparator)` will fix this problem too. But maybe it's better to rethink the whole method.

If we agree on that I can propose a fix with this two changes.

**Comment by jecisc** [comment]
For my part I agree. Do you want to try to enable appveyor build also?

**Comment by gcotelli** [comment]
I've also tested the changes locally in a Windows machine, and the deployment works creating the necessary subfolders

**Comment by theseion** [comment]
Looks ok to me.

**Comment by marschall** [comment]
Closed by #957
